syntax = "proto3";
package helloworld;

service Greeter{

rpc SayHello_v1 (HelloRequest) returns (stream HelloReply) {}
rpc SayHello_v2 (stream HelloRequest) returns (HelloReply) {}
rpc SayHello_v3 (stream HelloRequest) returns (stream HelloReply) {}

}

message HelloRequest {
string name =1;
}

message HelloReply {
string message = 1;
}